### PR TITLE
feat: add share position modal

### LINKED ISSUE #853: Implement Share Position Card
### Subject of the issue
Implement a Share button that when the user has an open position, the user can click on it, see the PnL graphic and share it through social media with their peers. 

### Related tickets
implements #728 
closes #469 

### Design
See Figma (shared by devDAO PM)

- Replace "Close at *date/time*" with "Created on *date/time*", showing the date and time the screenshot was generated
- Replace Avg Close Price" with "Current Price", showing the current oracle price at the time the screenshot was generated

#### Share button design
Screen Shot 2022-05-17 at 16 57 52

### Deliverables
- Add a Share button, incl. icon
- Make sure the share button is by default disabled. It is only enabled if a user has an open position in a given market.
- Upon click, open the "share modal". 
- The Share modal should include the pre-filled PnL graphic with current data of the open position
- Buttons to easily share the graphic via Twitter or via link to other services (e.g. on mobile messengers, email, etc.), and copy graphic to clipboard

### Deadline
Tuesday 24 May, 2022

### Rewards
150 SNX

### PR DESCRIPTION
**Pull request by 0xwontonsoup**
## Description
Closes #853

### CODE REVIEW

**Review by platschi — APPROVED** [review]
Tested and works great and as expected!

File: sections/futures/ShareModal/AmountContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+const lineSeparatorStyle = { margin: '0px 0.7vw 0px 0.7vw' };
+const currencyIconStyle = {
+	height: '1.94vw',
+	width: 'auto',
+	margin: '-0.3vw 0.5vw 0vw 0vw',
+};
+
+const AmountContainer: FC<AmountContainerProps> = ({ marketAsset, position }) => {
+	const marketAsset__RemovedSChar = marketAsset[0] === 's' ? marketAsset.slice(1) : marketAsset;
```
Comment (koredefashokun) [inline sections/futures/ShareModal/AmountContainer.tsx:23]: I think we can use `getDisplayAsset` here instead.

File: sections/futures/ShareModal/AmountContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+		}
+	};
+
+	return (
+		<>
+			<Container>
+				<StyledPositionType>
+					<CurrencyIcon
+						style={currencyIconStyle}
+						currencyKey={(marketAsset[0] !== 's' ? 's' : '') + marketAsset}
```
Comment (koredefashokun) [inline sections/futures/ShareModal/AmountContainer.tsx:45]: I think we can use `getMarketKeyFromAsset` or something here as well.

File: sections/futures/ShareModal/AmountContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+	color: #ffff;
+`;
+
+const StyledPositionType = styled.div`
+	display: flex;
+	flex-direction: row;
+`;
+
+/** @todo Import and use correct fonts from Figma design */
```
Comment (koredefashokun) [inline sections/futures/ShareModal/AmountContainer.tsx:90]: Not sure if this todo is still useful.

File: sections/futures/UserInfo/UserInfo.tsx [hunk 4]
```diff
@@ -65,6 +69,8 @@ const UserInfo: React.FC<UserInfoProps> = ({ marketAsset, position, openOrders,
 	const futuresPositionQuery = useGetFuturesPositionForAccount();
 	const futuresPositionHistory = futuresPositionQuery?.data ?? [];
```
Comment (koredefashokun) [inline sections/futures/UserInfo/UserInfo.tsx:70]: I think there's an ESLint warning on this line that we can fix.

**Review by koredefashokun — COMMENTED** [review]
Just left a few small comments.

File: styles/main.css [hunk 5]
```diff
@@ -37,6 +37,14 @@
 	font-display: monospace;
 }
 
+@font-face {
+	font-family: 'GT-America-Compressed-Medium';
+	src: url('/fonts/GT-America-Compressed-Medium.woff2') format('woff2'),
+		url('/fonts/GT-America-Compressed-Medium.woff') format('woff');
+	font-display: monospace;
```
Comment (asherism) [inline styles/main.css:44]: i don't think this typeface is monospaced

**Review by koredefashokun — APPROVED** [review]
Looks good.

**Review by platschi — APPROVED** [review]

**Pull request merged**